src/dal/models/polygon.ts

```diff
@@ -21,41 +22,58 @@ export interface Polygon extends ExampleEntitiesBase {
     borderWidth?: number;
 }
 
-export const polygonSchemaCreator: SchemaCreator = ((refNameCreator) => new Schema<Polygon>({
-    ...ExampleEntitiesBaseSchema,
-    positions: {
-        type: [[Number]],
-        required: true,
-    },
-    height: {
-        type: Number,
-        required: true,
-    },
-    comment: String,
-    aids: {
-        type: [String],
-        default: [],
-        required: true,
-    },
-    classification: [{
-        type: Schema.Types.ObjectId,
-        ref: refNameCreator(modelNames.MISSION),
-    }],
-    entityVersion: {
-        type: Number,
-        required: true,
-        default: 1,
-    },
-    expirationDate: Date,
-    internalMargins: Number,
-    externalMargins: Number,
-    fillColor: String,
-    entityColor: String,
-    borderStyle: {
-        type: Number,
-        required: false,
-    },
-    borderWidth: Number,
-}));
+export const polygonSchemaCreator: SchemaCreator = ((refNameCreator) => {
+    const schema = new Schema<Polygon>({
+        ...ExampleEntitiesBaseSchema,
+        entityName: {
+            type: String,
+            required: true,
+            default: 'polygon',
+        },
+        correlationId: {
+            type: String,
+            required: true,
+            default: () => getCurrId('polygon'),
+        },
+        positions: {
+            type: [[Number]],
+            required: true,
+        },
+        height: {
+            type: Number,
+            required: true,
+        },
+        comment: String,
+        aids: {
+            type: [String],
+            default: [],
+            required: true,
+        },
+        classification: [{
+            type: Schema.Types.ObjectId,
+            ref: refNameCreator(modelNames.MISSION),
+        }],
+        entityVersion: {
+            type: Number,
+            required: true,
+            default: 1,
+        },
+        expirationDate: Date,
+        internalMargins: Number,
+        externalMargins: Number,
+        fillColor: String,
+        entityColor: String,
+        borderStyle: {
+            type: Number,
+            required: false,
+        },
+        borderWidth: Number,
+    });
+    schema.pre('save', function(this: any, next: any) {
```

Comment:
Add comment that explains this is temporary situtaion.
TODO: Remove this when there would be solution to DataVersions. Either way, remove this shit in one month.